### ISSUE TITLE
npm start fails with 'TypeError:Reduce of empty array with no initial value'

### ISSUE THREAD

**Issue by JHaig343** [UoGSOCIS/website#27]
**Environment:**
Windows 10, version 1809
Node v. 10.16.0
NPM v. 6.9.0

**Steps to Reproduce:**
run `npm start` in the root directory
**Screenshot:**
![SOCIS Site Build Error](https://user-images.githubusercontent.com/24945283/58764914-0251f100-853b-11e9-8eb8-be7877976644.PNG)
On a closer look it seems like an issue with parsing the path; the path argument to shortPath I get is just 'C', which is popped into file so parts becomes an empty array. Might this be because I am working on Windows?

**Comment by MarshallAsch** [comment]
Yup that seems to be a windows issue, it was able to be reproduced on windows but not mac or linux. I will look into finding a work around

**Comment by MarshallAsch** [comment]
@JHaig343 Try running [this branch](https://github.com/UoGSOCIS/website/compare/27-windows-path-bug), I belive I have a fix there, the CI stuff is failing on windows due to NPM issues .

**Comment by JHaig343** [comment]
Unfortunately it didn't solve the issue, but it helped me figure out what the problem is.
the path argument to shortPath was just 'C' in the first place, so any splitting there was failing. Digging around, the culprit seems to be this line in parseStackFrame:
![Windows Build Problem](https://user-images.githubusercontent.com/24945283/58773172-09542000-858a-11e9-9f3e-812a4c49d821.PNG)
for reference, an Example Windows path returned by the logger is `C:\Users\blah\Documents\example.c:28:08`. The line above will split on the 1st colon and send 'C' to shortPath.
Digging around I found this as a potential solution:
![Windows Build Solution](https://user-images.githubusercontent.com/24945283/58773322-9ac39200-858a-11e9-9cd3-39a709e6fa81.PNG)
It splits on the 2nd semicolon and separates the path and line number, which fixes the error I was getting. It doesn't get the last part(column number I think?) of the log path so there is room for improvement, but it fixes the issue of `npm start` failing.
Also, I noticed that parts wasn't assigned in shortPath in the win32 system case, so undefined errors were popping up when trying to pop.
![Missing Parts Assign](https://user-images.githubusercontent.com/24945283/58773657-ec205100-858b-11e9-9864-0fee0d87c288.PNG)
Thanks for looking into the issue, I hope my solution here helps a bit!
